### DISCUSSION TITLE
Query vulnerabilities newer than certain datetime

### DISCUSSION THREAD

**Discussion by ptelang** [google/osv.dev#2065] [Q&A] [answered]
We would like to query all new vulnerabilities that have been reported since a given datetime? I believe it is not possible with the current APIs. Can you please confirm?

**Comment by oliverchang** [comment]
Hi, thank you for the question.

That's correct, we don't have such an API available.  Would you be able to help us understand the use case for this a bit more?

**Reply by ptelang** [reply]
We have a large database of packages. If any package in our DB is found to be malicious in OSV, we would like to mark it malicious in our DB. We can call osv API periodically (say 1hr) to fetch newly found vulnerabilities during that hour, and update our database. 

It will be even better if you can provide a real-time feed of malicious packages. That way we can subscribe to it instead of repeatedly calling the API.

**Reply by oliverchang** [reply, marked as answer]
Thanks for explanation. Would our bulk data dumps work for this use case? https://google.github.io/osv.dev/data/#data-dumps 

The volume of data is likely not going to be enough to warrant significant savings of an iterative fetch over a bulk download.

**Reply by ptelang** [reply]
Thanks, we will use the data dumps.
